python/cudf/cudf/dataframe/series.py

```diff
@@ -1795,6 +1801,52 @@ def round(self, decimals=0):
             index=self.index,
         )
 
+    def isin(self, test):
+
+        from cudf import DataFrame
+
+        lhs = self
+        rhs = None
+
+        try:
+            rhs = Series(columnops.as_column(test, dtype=self.dtype))
+        except Exception:
+            # pandas functionally returns all False when cleansing via
+            # typecasting fails
+            return Series(cudautils.zeros(len(self), dtype="bool"))
+
+        # If categorical, combine categories first
+        if pd.api.types.is_categorical_dtype(lhs):
```

Comment:
We're going to build a utility to streamline this more in the future, but this function is actually slow because it handles array-like's unnecessarily.

```suggestion
        if pd.api.types.pandas_dtype(lhs.dtype).type is pd.core.dtypes.dtypes.CategoricalDtypeType:
```

Reply:
I wouldn't mind standardizing these checks in this PR. I went through a few rounds of merge-from-upstream-then-revert-upstream-changes that kept breaking tests in this branch.

Reply:
If you don't mind that would be nice!

Reply:
done!